pysd/py_backend/model.py

```diff
@@ -1420,93 +1423,247 @@ def run(self, params=None, return_columns=None, return_timestamps=None,
         if reload:
             self.reload()
 
-        self.time.add_return_timestamps(return_timestamps)
-        if self.time.return_timestamps is not None and not final_time:
-            # if not final time given the model will end in the list
-            # return timestamp (the list is reversed for popping)
-            if self.time.return_timestamps:
-                final_time = self.time.return_timestamps[0]
-            else:
-                final_time = self.time.next_return
+        self._config_simulation(params, return_columns, return_timestamps,
+                                initial_condition, final_time, time_step,
+                                saveper, cache_output, progress=progress)
 
-        self.time.set_control_vars(
-            final_time=final_time, time_step=time_step, saveper=saveper)
+        # instante output object
+        self.output = ModelOutput(output_file)
+        self.output.set_capture_elements(self.capture_elements)
+        self.output.initialize(self)
+
+        self._integrate()
+
```

Comment:
constant cache should be removed

Reply:
also del self._dependencies["OUTPUTS"]

Reply:
Both actions should be done also in the step run whenever the step run is finished

Reply:
Sorry I just saw that this is done in the collect method of the output